pkg/agent/executor/exec_monitor.go

```diff
@@ -107,6 +107,7 @@ func (c *ExecutorCoreWithClients) DoMonitorDownscales(ctx context.Context, logge
 				logger.Info("vm-monitor approved downscale", logFields...)
 				if unchanged {
 					state.Monitor().DownscaleRequestAllowed(endTime)
+					state.Monitor().UpdateLogicalTime(action.DesiredLogicalTime.Rewind(endTime))
```

Comment:
I noticed we have a these new `UpdateLogicalTime` methods. Can they be merged with the response handling methods instead? (e.g., just passing `endTime` and `action.DesiredLogicalTime` to `.Monitor().DownscaleRequestAllowed()` ?)

Reply:
I tried that, and the diff was too much, especially in tests.

Reply:
I tried it in 896d0df8, a little ugly, but manageable. WDYT?

Reply:
Fair enough — testing locally, the following also works:

```diff
diff --git a/pkg/agent/core/state_test.go b/pkg/agent/core/state_test.go
index f019e4c5..26f26354 100644
--- a/pkg/agent/core/state_test.go
+++ b/pkg/agent/core/state_test.go
@@ -19,7 +19,7 @@ import (
 	"github.com/neondatabase/autoscaling/pkg/api"
 )
 
-var NilLogicalTime = helpers.SafeVal[vmv1.LogicalTime](nil)
+var NilLogicalTime = lo.Empty[*vmv1.LogicalTime]()
 
 func Test_DesiredResourcesFromMetricsOrRequestedUpscaling(t *testing.T) {
 	slotSize := api.Bytes(1 << 30 /* 1 Gi */)
```

(I remember putting a lot of effort into making sure that the nil-handling in testhelpers worked properly :sweat_smile:)